src/core/config.go

```diff
@@ -429,6 +429,7 @@ type Configuration struct {
 		CgoCCTool     string `help:"Sets the location of CC while building cgo_library and cgo_test rules. Defaults to gcc" var:"CGO_CC_TOOL"`
 		FilterTool    string `help:"Sets the location of the please_go_filter tool that is used to filter source files against build constraints." var:"GO_FILTER_TOOL"`
 		DefaultStatic bool   `help:"Sets Go binaries to default to static linking. Note that enabling this may have negative consequences for some code, including Go's DNS lookup code in the net module." var:"GO_DEFAULT_STATIC"`
+		GoBuildCompat bool   `help:"Changes the behavior of the build rules to be more compatible with go build i.e. please will descend into the package directory to run unit tests so go test does." var:"GO_BUILD_COMPAT"`
```

Comment:
Can we think of a more specific name than this? It only does one aspect of go build compatibility (I guess `GoTestCompat` would be a bit closer)